### ISSUE TITLE
Project settings are ignored in PEP 723 scripts

### ISSUE THREAD

**Issue by marcelwa** [astral-sh/ty#4083]
### Summary

Starting in `v0.0.62`, `ty check` ignores `[tool.ty.analysis].allowed-unresolved-imports` configured in `pyproject.toml`.

### Reproduction

1. Configure `pyproject.toml` with:
```toml
[tool.ty.analysis]
allowed-unresolved-imports = ["some_external_lib.**"]
```

2. Create a file `example.py`:
```python
from some_external_lib.submodule import helper
```

3. Run `ty check`.

### Expected Result
Unresolved import diagnostics matching the glob are suppressed (as they were in `v0.0.61` and earlier).

### Actual Result
`v0.0.62` and `v0.0.63` emit `error[unresolved-import]: Cannot resolve imported module 'some_external_lib.submodule'`.

### Version
`v0.0.62`, `v0.0.63` (worked as expected in `v0.0.56` – `v0.0.61`).

**Comment by MichaReiser** [comment]
I suspect that this is related to the PEP 723 script changes (https://github.com/astral-sh/ruff/pull/26671). Does the setting not work in any python file or is it only in Python files that have a PEP 723 script header?

**Comment by marcelwa** [comment]
Ah, good catch, that's exactly it. Tested this with a quick repro: `allowed-unresolved-imports` in `pyproject.toml` is still respected for standard Python files in `v0.0.63`, but starting in `v0.0.62` it gets ignored on files with a PEP 723 script header (`# /// script`).

### Minimal Repro

`pyproject.toml`:
```toml
[project]
name = "demo"
version = "0.1.0"
requires-python = ">=3.10"

[tool.ty.analysis]
allowed-unresolved-imports = ["some_uninstalled_module.**"]
```

`script.py`:
```python
# /// script
# requires-python = ">=3.10"
# dependencies = []
# ///
from some_uninstalled_module import foo
```

* `ty@0.0.61 check script.py` → **Passes**
* `ty@0.0.63 check script.py` → **Fails** (`error[unresolved-import]`)

**Comment by MichaReiser** [comment]
This behavior change was intentional. The idea behind it is that scripts are really their own project and they should work in isolation. Can you say more about your use case? What behavior would you expect and why?

**Comment by marcelwa** [comment]
Our main use case is executable demonstration scripts inside an `examples/` directory in a repository. They include PEP 723 inline metadata (`# /// script`) so users can easily run them standalone with `uv run`.

When running project-wide type checking in pre-commit or CI (`ty check` at the repo root), we check all Python files in the repo. Some example scripts import optional heavy packages that we deliberately omit from the core project/dev dependencies to keep the environment lean, so we configure `[tool.ty.analysis].allowed-unresolved-imports` in the root `pyproject.toml`.

Because `ty check` now treats PEP 723 scripts as isolated, project-level settings like `allowed-unresolved-imports` are ignored for those files, causing project-wide `ty check` / pre-commit to fail.

### What behavior we'd expect

1. When running repository-wide checks (`ty check` from the project root), files within the repository should inherit analysis settings like `allowed-unresolved-imports` or `exclude` from the root `pyproject.toml`.
2. Alternatively, allowing `[tool.ty.analysis]` sections directly inside the PEP 723 `# /// script` metadata block so standalone scripts can declare their own type-checker suppressions.

**Comment by MichaReiser** [comment]
`[tool.ty.analysis]` this might already work. If not, it's what I'm currently working on (allow scripts to change any ty setting). 

Another question, since I haven't had much chance to talk to script users and your feedback is very valuable on this. We plan to automatically sync the virtual environment for scripts and check them in full isolation. Would you expect ty to check all scripts by default or would you prefer an opt-in, e.g. by using `ty check --all-scripts`?

**Comment by marcelwa** [comment]
Just tested it in `v0.0.63`, adding `[tool.ty.analysis]` directly inside the `# /// script` block already works! That solves our immediate configuration need for standalone scripts.

Regarding your question on auto-syncing script venvs during `ty check`:

From a developer/CI perspective, when running `ty check` at the project root, the expectation is that all Python files in the repo get checked so broken scripts are caught before merging.

However, if auto-syncing isolated venvs for scripts involves network requests or resolving heavy script dependencies (like PyTorch or PySCF), doing that automatically during quick local runs or pre-commit hooks could slow down the dev loop.

A great balance might be:
- Check all scripts by default if syncing is cached/fast, with a flag (e.g. `--no-scripts` or a config setting) to bypass isolated script syncs when speed is critical.
- Or provide an explicit `ty check --all-scripts` (or pre-commit hook variant) so repos with heavy script dependencies can choose whether to run isolated script checks locally or defer them to CI.

What do you think?

**Comment by MichaReiser** [comment]
Thank you. That's very valuable. This roughly matches our thinking. To me, the open question is what's less surprising:

* `ty check` being slow because it checks and syncs all scripts by default
* Missing a broken script because checking all scripts (not `ty check script.py`) requires explicit opt-in

The second one seems worse to me, as it can easily go unnoticed. You notice the first one and can disable it.

**Comment by marcelwa** [comment]
Checking all scripts by default definitely feels like the less surprising and safer choice to me. Missing a broken script is a silent failure that potentially lands in `main` and hits users, whereas a slower initial check is immediately obvious and actionable. So yeah, I agree with you.

Since `ty` builds on `uv`'s caching, that environment setup cost should really only happen on the first run anyway. As long as there's clear CLI feedback when dependencies are being fetched for a script, developers will understand what's happening (and can opt out if they really want to).

Flags like `--no-sync` or `--cached-only` (or a pyproject setting) would be a nice escape hatch for offline work or tight local dev loops.
